docs/manifests_and_customizing_via_api.md

```diff
@@ -0,0 +1,341 @@
+
+At this time you must run `kops create cluster` and then export the YAML from the state store.   We plan in the future to have the capability to generate kops YAML via the command line.  The following is an example of creating a cluster and exporting the YAML.
+
+```
+export NAME=k8s.example.com
+export KOPS_STATE_STORE=s3://example-state-store
+ kops create cluster $NAME \
+    --zones "us-east-2a,us-east-2b,us-east-2c" \
+    --master-zones "us-east-2a,us-east-2b,us-east-2c" \
+    --networking weave \
+    --topology private \
+    --bastion \
+    --node-count 3 \
+    --node-size m4.xlarge \
+    --kubernetes-version v1.6.6 \
+    --master-size m4.large \
+    --vpc vpc-6335dd1a
+```
+
+The next step is to export the configuration to a YAML document.  `kops` has a command that allows the export in a single YAML document, but since JSON files need to separate documents, we only export YAML with a single command.  You can export JSON with multiple commands.
+
+```
+kops get $NAME -o yaml > $NAME.yaml
+```
+
+The above command exports a YAML document which contains the definition of the cluster, `kind: Cluster`, and the definitions of the instance groups, `kind: InstanceGroup`.
+
+The following is the contents of the exported YAML file.
+
+```YAML
+apiVersion: kops/v1alpha2
+kind: Cluster
+metadata:
+  creationTimestamp: 2017-05-04T23:21:47Z
+  name: k8s.example.com
+spec:
+  api:
+    loadBalancer:
+      type: Public
+  authorization:
+    alwaysAllow: {}
+  channel: alpha
+  cloudProvider: aws
+  configBase: s3://example-state-store/k8s.example.com
+  etcdClusters:
+  - etcdMembers:
+    - instanceGroup: master-us-east-2d
+      name: a
+    - instanceGroup: master-us-east-2b
+      name: b
+    - instanceGroup: master-us-east-2c
+      name: c
+    name: main
+  - etcdMembers:
+    - instanceGroup: master-us-east-2d
+      name: a
+    - instanceGroup: master-us-east-2b
+      name: b
+    - instanceGroup: master-us-east-2c
+      name: c
+    name: events
+  kubernetesApiAccess:
+  - 0.0.0.0/0
+  kubernetesVersion: 1.6.6
+  masterPublicName: api.k8s.example.com
+  networkCIDR: 172.20.0.0/16
+  networkID: vpc-6335dd1a
+  networking:
+    weave: {}
+  nonMasqueradeCIDR: 100.64.0.0/10
+  sshAccess:
+  - 0.0.0.0/0
+  subnets:
+  - cidr: 172.20.32.0/19
+    name: us-east-2d
+    type: Private
+    zone: us-east-2d
+  - cidr: 172.20.64.0/19
+    name: us-east-2b
+    type: Private
+    zone: us-east-2b
+  - cidr: 172.20.96.0/19
+    name: us-east-2c
+    type: Private
+    zone: us-east-2c
+  - cidr: 172.20.0.0/22
+    name: utility-us-east-2d
+    type: Utility
+    zone: us-east-2d
+  - cidr: 172.20.4.0/22
+    name: utility-us-east-2b
+    type: Utility
+    zone: us-east-2b
+  - cidr: 172.20.8.0/22
+    name: utility-us-east-2c
+    type: Utility
+    zone: us-east-2c
+  topology:
+    bastion:
+      bastionPublicName: bastion.k8s.example.com
+    dns:
+      type: Public
+    masters: private
+    nodes: private
+
+---
+
```

Comment:
Since I imagine some people will use this to start with, a note about default disk size (or a bigger default than 20GB) would prevent some headaches.